### ISSUE TITLE
Rule Idea: selector-no-descendant-selector

### ISSUE THREAD

**Issue by kswilster** [kristerkari/stylelint-scss#97]
This is a new rule for preventing the use of descendant selectors. We want to completely eliminate the use of descendant selectors in our CSS, and the use of other PostCSS features makes them unnecessary.
Inspired by [Firefox UI Guidelines](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Writing_efficient_CSS#Avoid_the_descendant_selector.21).

```scss
//good
.foo.bar {}

.foo {
  &.bar {}
}

//bad
.foo .bar {}

.foo {
  .bar {}
}
```

**Comment by jeddy3** [comment]
@kswilster I think stylelint's [`selector-max-compound-selectors`](http://stylelint.io/user-guide/rules/selector-max-compound-selectors/) rule is what you're after. 

The rule resolves nested selectors and so the following config:

```
{
  "rules": {
    "selector-max-compound-selectors": 1
  }
}
```

Will warn for both:

```scss
.foo .bar {}

.foo {
  .bar {}
}
```

**Comment by dryoma** [comment]
Yup, the idea behind that rule is almost exactly that.

**Comment by dryoma** [comment]
Hi @kswilster 

Is the `selector-max-compound-selectors` mentioned above suit your needs, or is there still elements you would like implemented (regarding the initial proposal)?

**Comment by kswilster** [comment]
the `selector-max-compound-selectors` works out perfectly for me, thanks!
